src/protocol.vala

```diff
@@ -453,11 +504,22 @@ namespace LanguageServer {
         TypeParameter = 25
     }
 
+    class CompletionClientCapabilities : Object {
+        public class CompletionItemCapabilities : Object {
+            public bool snippetSupport { get; set; }
+        }
+
+        public bool dynamicRegistration { get; set; }
+        public CompletionItemCapabilities? completionItem { get; set; }
+
+    }
+
     class TextDocumentClientCapabilities : Object {
         public class DocumentSymbolCapabilities : Object {
             public bool hierarchicalDocumentSymbolSupport { get; set; }
         }
         public DocumentSymbolCapabilities documentSymbol { get; set; default = new DocumentSymbolCapabilities ();}
+        public CompletionClientCapabilities completion { get; set; }
```

Comment:
To keep the purpose of this merge commit clear, please don't change anything else that is not strictly pertinent to the issue.